src/v/config/property.h

```diff
@@ -568,6 +585,70 @@ class conversion_binding : public binding_base<T> {
     }
 };
 
+/**
+ * A property sanctioning binding contains a copy of the property's
+ * value, which will be updated in-place whenever the property is
+ * updated in the cluster configuration. It offers both free access
+ * to the value and a sanctioned view, in case the value is restricted.
+ *
+ * This is useful for classes that want a copy of a property without
+ * having to write their own logic for subscribing to value changes.
+ */
+template<class T>
+class sanctioning_binding : public binding_base<T> {
```

Comment:
Can this be done in layers depending on configuration rather than being a part of configuration. This is a feature that can easily be implemented elsewhere (maybe as a wrapper for binding)

Reply:
The issue with building `sanctioning_binding` around `binding` is that the latter is not exposing enough functionality to be an option.

We could always expand the `binding` with what it is missing (default value and restricting checking) to allow for it, but then we have a larger `binding` object that we don't need for the majority of cases.

Reply:
maybe a sanctioned binding can accept property pointer/reference and current binding ?